elliptic-curve/src/weierstrass.rs

```diff
@@ -0,0 +1,128 @@
+//! Complete projective formulas for prime order elliptic curves as described
+//! in [Renes-Costello-Batina 2015].
+//!
+//! [Renes-Costello-Batina 2015]: https://eprint.iacr.org/2015/1060
+
+#![allow(clippy::op_ref)]
+
+use ff::Field;
+
+/// Affine point whose coordinates are represented by the given field element.
+pub type AffinePoint<Fe> = (Fe, Fe);
+
+/// Projective point whose coordinates are represented by the given field element.
+pub type ProjectivePoint<Fe> = (Fe, Fe, Fe);
```

Comment:
The use of coordinate tuples is to prevent confusion with the `AffinePoint` and `ProjectivePoint` types currently defined on a crate-by-crate basis (i.e. in `p256` and `p384`)

Perhaps it would be possible in an eventual followup PR to extract a generic `ProjectivePoint` type which impls these formulas for `C: PrimeCurve`.